fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanItemReader.java

```diff
@@ -49,7 +49,7 @@ public void beforeStep(@NotNull StepExecution stepExecution) {
             loanIds = Collections.emptyList();
         } else {
             loanIds = loanRepository.findAllNonClosedLoansBehindOrNullByMinAndMaxLoanId(loanCOBParameter.getMinLoanId(),
-                    loanCOBParameter.getMaxLoanId(), ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE));
+                    loanCOBParameter.getMaxLoanId(), ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE).minusDays(1));
```

Comment:
why -1 day?

Reply:
Also this one is not correct as it is ignoring any already locked loans (locked by INLINE COB for example)

Reply:
It is ignoring if the loan is already locked and maybe failed...maybe it failed on the very first execution and the closed business date is still NULL, but it failed 2 days ago... it is edge case, but we need to be careful here...(probably not cause any issues in the other hand...)

Reply:
moved the `NUMBER_OF_DAYS_BEHIND` to `LoanCOBConstant` and use it here as well